docs/proposals/configuration.md

```diff
@@ -0,0 +1,287 @@
+---
+title: Configuration Files Design
+
+authors:
+  - "@kadisi"
+  - "@fisherxu"
+  
+approvers:
+  - "@kevin-wangzefeng"
+  - "@sids-b"
+
+creation-date: 2019-10-02
+
+---
+
+* [Configuration Files Design](#configuration-files-design)
+   * [Abstract](#abstract)
+   * [Goals](#goals)
+   * [Proposal](#proposal)
+      * [cloudcore](#cloudcore)
+         * [config struct](#config-struct)
+         * [default config file](#default-config-file)
+      * [edgecore](#edgecore)
+         * [config struct](#config-struct-1)
+         * [default config file](#default-config-file-1)
+
+# Configuration Files Design
+
+
+## Abstract  
+
+Currently, cloudcore/edgecore/edgesite\`s configuration files are in the conf directory at the same level and have 3 configuration files, it is difficult to maintain and extend.
+
+Now KubeEdge use the `beehive` package to analyze configuration files, but it has many drawbacks,when the program is running, it will output a lot of logs first.
+
+Is also no need to read configuration variables in multiple ways, such as commandline and env.
+
+## Goals
+
+Binary default reads yaml files in `/etc/kubeedge/config` directory and then performs the relevant logical processing.
+
+When binary run, it will first load config yaml file by `gopkg.in/yaml.v2` package rather than `beehive` package,
+
+and it also need to easily generate the configuration files for `keadm`.
+
+
+## Proposal
+
+### cloudcore 
+
+#### config struct 
+
+```go
+
+type CloudCoreConfig struct {
+    Kube           *KubeConfig           `yaml:"kube"`
+    EdgeController *EdgeControllerConfig `yaml:"edgeController"`
```

Comment:
Leave an empty DeviceController here?  Although there are no configuration for it now

Reply:
ok